roles/cockpit/tasks/main.yml

```diff
@@ -0,0 +1,99 @@
+#########################################################################
+# Title:         Sandbox: Cockpit Role                                  #
+# Author(s):     CHAIR/Raneydazed                                       #
+# URL:           https://github.com/saltyorg/Sandbox                    #
+# --                                                                    #
+#########################################################################
+#                   GNU General Public License v3.0                     #
+#########################################################################
+---
+- name: Add DNS record
+  ansible.builtin.include_tasks: "{{ resources_tasks_path }}/dns/tasker.yml"
+  vars:
+    dns_record: "{{ lookup('vars', role_name + '_dns_record') }}"
+    dns_zone: "{{ lookup('vars', role_name + '_dns_zone') }}"
+    dns_proxy: "{{ lookup('vars', role_name + '_dns_proxy') }}"
+
+- name: Check if 'cockpit_socket' exists
+  ansible.builtin.stat:
+    path: "{{ cockpit_paths_socket_location }}"
+  register: cockpit_socket
+
+- name: Stop existing 'cockpit_socket'
+  ansible.builtin.systemd:
+    state: stopped
+    name: cockpit.socket
+  when: cockpit_socket.stat.exists
+
+# Need to add a task to start the cockpit service as well
+
+- name: Check if 'cockpit_service' exists
+  ansible.builtin.stat:
+    path: "{{ cockpit_paths_service_location }}"
+  register: cockpit_service
+
+- name: Stop existing 'cockpit_service'
+  ansible.builtin.systemd:
+    state: stopped
+    name: cockpit.service
+  when: cockpit_service.stat.exists
+
+# Also need to fix the traefik file to include the correct settings
+
+- name: Create directories
+  ansible.builtin.include_tasks: "{{ resources_tasks_path }}/directories/create_directories.yml"
+
+- name: Import 'cockpit_socket'
+  ansible.builtin.template:
+    src: listen.conf.j2
+    dest: "{{ cockpit_paths_socket_location }}"
+    mode: "0664"
+    force: true
+
+- name: Import 'cockpit_config'
+  ansible.builtin.template:
+    src: cockpit.conf.j2
+    dest: "{{ cockpit_paths_config_location }}"
+    mode: "0664"
+    force: true
+
+- name: Import 'cockpit_traefik'
+  ansible.builtin.template:
+    src: cockpit.yml.j2
+    dest: "{{ cockpit_paths_traefik_location }}"
+    mode: "0775"
+    force: true
+
+- name: Change traefik file ownership
+  ansible.builtin.file:
+    path: "{{ cockpit_paths_traefik_location }}"
+    owner: "{{ uid }}"
+    group: "{{ gid }}"
+  when:
+    - not continuous_integration
+    # - not cockpit_socket.stat.exists
+
+- name: Install Cockpit from backports
```

Comment:
Missing state, which should be latest if you ever want this to update and it should also freeze the apt package if you only want the role to manage updates.